boltdesignsystem/bolt
# Create new standalone Deprecation Modal UI to alert users of IE 11 being deprecated

## Jira
http://vjira2:8080/browse/BDS-2004

## Summary
Adds a new standalone "deprecation modal" to automatically alert users of IE 11 being deprecated (with the idea of repurposing this work for other use cases in the future). 

![CleanShot 2020-03-04 at 09 54 18](https://user-images.githubusercontent.com/1617209/75891726-31fd9480-5dfe-11ea-92f7-d2facc56b6f0.gif)

## Definition of Done

### Functionality:

- [ ] Automatically display in IE 11 via an IE 11-specific media query
- [ ] This notification should be fully embedded on the initial page (vs getting dynamically loaded) so we don't need to worry about external JS tripping this up.
- [ ] Handle the dismiss icon-only button via the checkbox / label pattern to try to gracefully handle JS errors that I guarantee will be thrown in IE 11. 
- [ ] Create a tiny amount of JS to try to handle "remembering" being dismissed (via cookie). Keep in mind that external JS errors might end up preventing this from working as expected.
- [ ] This should ideally be written in vanilla JS (transpiled through Babel)
- [ ] Using a tiny 3rd party library (ex. to handle cookies) is fine if absolutely necessary as long as we’re bundle all this into a single static HTML file 

### Styling:
- [ ] Fixed position modal 
- [ ] Doesn't use any of Bolt's Core's Sass functions / mixins.
- [ ] All styles written should use custom scoped BEM classes for styling this particular bit of UI — doesn't assume any external Bolt code (CSS / JS) exists! 

### Build / Bundling
- [ ] Automatically inline all assets used (CSS, JS, SVGs) via build. No external resources!
- [ ] Assets are bundled together via tiny standalone (Parcel) build. The idea here is to completely isolate this deliverable from the rest of the build process. Parcel automatically inlines the compiled Sass and JS icon into the HTML file that gets output and published to NPM + pointed to with Twig.

### General / Misc
- [ ] Inline static SVG icon 
- [ ] We shouldn't be using any existing Bolt components to build this out (Text, link, button, icon, inline list... none!) so no polyfills, no external assets or external dependencies are required.
- [ ] The tiny amount of content that shows up in here is hard-coded for max simplicity

<hr>

- [ ]  Ideal Deliverable: this IE 11 deprecation notification gets compiled out as an HTML file with auto-inlined JS / CSS that Twig can then include.

## How to test
- Review DOD from above matches up with this PR
- Test UI in IE 11 to confirm the cookie functionality, UI itself all working as expected
- Confirm E2E integration tests passing as expected
